### ISSUE TITLE
IndexError: too many indices for array

### ISSUE THREAD

**Issue by deric** [ugursogukpinar/csv2arff#3]
I have a CSV with 302 lines (and many columns), and I'm getting this error:

```
Reading csv file to convert arff file
Calculating attribute types
Traceback (most recent call last):
  File "/usr/local/bin/csv2arff", line 9, in <module>
    load_entry_point('csv2arff==1.4', 'console_scripts', 'csv2arff')()
  File "/usr/local/lib/python2.7/dist-packages/scripts/csv2arff.py", line 100, in main
    Csv2Arff(sys.argv[1], sys.argv[2])
  File "/usr/local/lib/python2.7/dist-packages/scripts/csv2arff.py", line 16, in __init__
    self.determine_attribute_types()
  File "/usr/local/lib/python2.7/dist-packages/scripts/csv2arff.py", line 28, in determine_attribute_types
    unique_values = list(set(self.data[:,i]))
IndexError: too many indices for array
```

**Comment by ugursogukpinar** [comment]
I tried with this dataset as example and it works well. Is there any chance to see your dataset?
[train.csv.zip](https://github.com/ugursogukpinar/csv2arff/files/533359/train.csv.zip)

**Comment by deric** [comment]
Yeah, sure. I guess that the problem is causing too many columns (1001). Here's [the file](https://ulozto.net/!70vEZ7YnVNeK/single-cell-mrna-pollen-csv).

**Comment by deric** [comment]
It's rather strange issue. Just [first two lines](http://pastebin.com/UXzr5dPV) of the file are enough to reproduce the issue.

While with other files an array of `ndarray`s is returned:

```
 columns: 1001

 data size: 223

 data shape: (223, 1001)

 data [['ENSMUSG00000022037' 'ENSMUSG00000017390' 'ENSMUSG00000006205' ...,
  'ENSMUSG00000026254' 'ENSMUSG00000021018' 'cell']
 ['1.5053828491' '1.118359726' '7.900758149' ..., '0' '0' 'GP']
 ['6.5028079412' '6.321603452' '9.0340616107' ..., '6.2896685014' '0' 'GP']
 ..., 
 ['0.9237196787' '0' '0' ..., '0' '0' 'PSA']
 ['0' '0' '0' ..., '0.0894981508' '5.1495032631' 'PSA']
 ['0.2868811478' '0.6672109125' '0' ..., '0' '0' 'PSA']]

 row 0 <type 'numpy.ndarray'>
 row 1 <type 'numpy.ndarray'>
```

in case of the "broken" CSV it returns arrya of `numpy.void`:

```

 columns: 1001

 data size: 1

 data shape: (1,)
..., '7.6433511581', '5.2452674665', '7.4459252986', '3.2675357981', '3.9439213266', '6.2818836746', '7.3309168781', '0.3103401206', '4.5484366247', '4.7818842345', '0.5360529002', '7.2752864681', '4.3840498068', '4.2418401836', '6.0274637234', '2.2779847473', '7.9729223642', '7.7241045199', '4.1513717763', '6.662632775', '0', '5.8414699995', '8.464668267', '5.413458688', '8.8173674378', '3.408711861', '6.1201860275', '4.4449320489', '5.4419481905', '5.866166169', '5.5597976958', '5.0465783666', '1.3219280949', '8.0905888274', '0', '0', '6.5470487506', '6.3183168413', '5.7117702039', '6.0513721017', '0', '2338')]
 row 0 <type 'numpy.void'>
 row 1 <type 'numpy.void'>
```

**Comment by deric** [comment]
fixed during refactoring in a332ca936d0f557cecf6659958e86c3b7e80b146
